src/Symfony/Bundle/WebProfilerBundle/Controller/ProfilerController.php

```diff
@@ -193,8 +194,8 @@ public function toolbarAction(Request $request, $token)
         $url = null;
         try {
             $url = $this->generator->generate('_profiler', array('token' => $token));
-        } catch (\Exception $e) {
-            // the profiler is not enabled
+        } catch (RouteNotFoundException $e) {
+            // the named route doesn't exist => the profiler is not enabled
```

Comment:
We now don't catch other exceptions anymore (which might happen when the `_profiler` route is overridden).